docs/guides/rendering.md

```diff
@@ -1,445 +1,350 @@
+
+But as the details of this document reveal, rendering in napari is, in fact, very complicated.
+
+Consider some of the more important reasons for this.
+
+- Multiple layers can have different extents with different transforms.
+- Different layer types (e.g. Images vs. Points) handle slicing differently.
+- Layer data can be large or slow to load into RAM.
+- Sliced data in VRAM may exceed the maximum texture size.
+- There are experimental settings that enable asynchronous slicing.
+
+As a result, rendering in napari is the source of many bugs and performance problems that we are actively trying to fix and improve.
+
+This document describes napari's simple rendering paths with pointers to the more powerful, unusual, and complicated ones.
+
+We will use scikit-image's 3D cells data as a running example throughout this documentation.
+
+```{code-cell} python
+import napari
+
+viewer = napari.Viewer()
+viewer.open_sample('napari', 'cells3d')
+```
+
+```{code-cell} python
+:tags: [hide-input]
+from napari.utils import nbscreenshot
+
+nbscreenshot(viewer, alt_text="3D cell nuclei and membranes rendered as 2D slices in the napari viewer")
+```
+
+## Dimensions
+
+The region visible in napari's canvas is almost entirely determined by the state in {attr}`ViewerModel.dims<napari.components.ViewerModel.dims>`.
+Changes to the attributes and properties of this class typically represent changes to that region, which then triggers slicing and rendering
+so that the canvas is updated to present layers' data in that region.
+
+### Range and world extent
+
+{attr}`Dims.range<napari.components.Dims.range>` describes the extent of all layers in their shared world coordinate system,
+in addition to the `step` that should be taken in each dimension as its corresponding slider position changes.
+
+In our running example, the range
+
+```{code-cell} python
+viewer.dims.range
+```
+
+is solely determined by the shape of the data
+
+```{code-cell} python
+viewer.layers[0].data.shape
+```
+
+because the layers have the same shape and identity transforms.
+
+### Point and selective slicing
+
+{attr}`Dims.point<napari.components.Dims.point>` describes the coordinates of the current slice plane in that same world coordinate system.
+
+In our running example, the default 2D view defines {attr}`Dims.point<napari.components.Dims.point>` to be
+
+```{code-cell} python
+viewer.dims.point
+```
+
+which represents the mid-point through all three dimensions.
+As the last two dimensions are visualized in the canvas, this represents the 2D plane through the middle of the first dimension.
+
+Only the dimensions in {attr}`Dims.not_displayed<napari.components.Dims.not_displayed>` have meaningful values in
+{attr}`Dims.point<napari.components.Dims.point>` because all data in displayed dimensions is retained in a slice,
+even though those data may not be visible in the canvas due to the current camera parameter values.
```

Comment:
so this is to say that even if I'm zoomed in and can only see a part of the current 2D slice, all the data in the slice is still loaded, right?

Reply:
Yes, unless it's multiscale/tiled.